src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,92 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_SIZE = 16;
+    private static final int RESIZE = 2;
+    private static final double THRESHOLD = 0.75;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        this.table = new Node[DEFAULT_SIZE];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        checkForResize();
+        Node<K, V> newNode = new Node<>(key, value);
+        int index = getIndex(key);
+        Node<K, V> node = table[index];
+        if (node == null) {
+            table[index] = newNode;
+            size++;
+        } else {
+            while (node != null) {
+                if (Objects.equals(key, node.key)) {
+                    node.value = value;
+                    return;
+                } else if (node.next == null) {
+                    node.next = newNode;
+                    size++;
+                    return;
+                }
+                node = node.next;
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if (Objects.equals(key, node.key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getIndex(K key) {
+        return key == null ? 0 : Math.abs(key.hashCode() % table.length);
+    }
+
+    private void checkForResize() {
+        if (size == THRESHOLD * table.length) {
+            resize();
+        }
+    }
```

Comment:
maybe you can relocate the logic of this method to your resize method, also as I know, the resize should me invoked if size > threshold, not ==